geps/gep-1651.md

```diff
@@ -0,0 +1,154 @@
+# GEP-1651: GatewayAddress Scope
+
+* Issue: [#1651](https://github.com/kubernetes-sigs/gateway-api/issues/1651)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## TLDR
+
+Allow users to configure a GatewayAddress so that it is only accessible within
+a specific scope (ie. public/cluster local)
+
+## Goals
+
+- Cluster local gateways can be created in a declarative way `ClusterLocal` and
+  are only accessible within a single cluster
+- Define a `Public` scope for addresses that are accessible to the internet
+- Implementations can define & support their own scopes
+
+## Non-Goals
+
+- Per-request/route scope
+- Not a lightweight service mesh
+
+## Introduction
+
+One of the early feature requests for Knative was the ability to deploy an 
+application using Knative's HTTP routing support, but make it only available
+within the cluster. I want to be able to specify both the "internal" 
+(service.namespace.svc) and "external" (service.namespace.example.com). 
+Gateways using the same GatewayClass on the cluster, but ensure that the 
+"internal" service is only reachable within the cluster. This would greatly 
+simplify deployment for users over the instructions we have today.
+
+The reason why we require an "internal" Gateway is so we can take advantage of 
+layer 7 load balancing for our internal services.
+
+## API
+
+We propose extending the `GatewayAddress` type to allow end-users to specify
+the accessibility of the requested address using a property named `Scope`. 
+
+`GatewayAddress.Value` is now an `optional` field. This allows users to signal
```

Comment:
This would be a breaking change.

Reply:
I'm confused how is this breaking? Existing clients that specify an address value would continue to work.

Generally I view making required fields optional non-breaking.

Reply:
My bad... required > optional is NOT a breaking change.

Reply:
Actually in #1596, required -> optional was cited as a breaking change because implementations would have to start accommodating the absence of that field and could no longer assume that the field would be populatedf

Reply:
Although we technically leave room for "making required fields optional" in minor versions (https://gateway-api.sigs.k8s.io/concepts/versioning/#minor-version-eg-v040-v050), I agree that it could result in breaking implementations. I think that was a very high risk with making listeners optional. Wherever we can, I'd like to avoid this transition, or at the very least, announce it _well in advance_ of making the change.

Reply:
This is removed from the GEP already